internal/network/resource_subnet.go

```diff
@@ -168,7 +169,8 @@ func resourceNetworkSubnetDelete(ctx context.Context, d *schema.ResourceData, m
 		if hcloud.IsError(err, hcloud.ErrorCodeConflict) || hcloud.IsError(err, hcloud.ErrorCodeLocked) {
 			return err
 		}
-		if hcloud.IsError(err, hcloud.ErrorCodeServiceError) && strings.Contains(err.Error(), "servers are attached") {
+		if hcloud.IsError(err, hcloud.ErrorCodeServiceError) &&
+			(strings.Contains(err.Error(), "servers are attached") || strings.Contains(err.Error(), "network has attached resources")) {
```

Comment:
This feels dirty, any plan to handle this with a not human/changeable error message.